core/src/lib/page-object-builder.ts

```diff
@@ -56,61 +40,21 @@ export class PageObjectBuilder {
 
     /**
      * Creates an instance of PageObjectBuilder.
-     * @param {{
-     *             codeBuilder?: QueuedCodeBuilder,
-     *             awaiter?: Awaiter,
-     *             e2eTestPath?: string,
-     *             waitForAngularEnabled?: boolean,
-     *         }} params
+     * @param {IPageObjectBuilderOptions} options
      * @memberof PageObjectBuilder
      */
     constructor(
-        params: {
-            /**
-             * Defines the CodeBuilder. See more at QueuedCodeBuilder
-             *
-             * @type {QueuedCodeBuilder}
-             * @default new QueuedCodeBuilder('  ')
-             */
-            codeBuilder?: QueuedCodeBuilder,
-            /**
-             * Action that should be performed normally. See more at IGenerationInstruct
-             *
-             * @type {Awaiter}
-             * @default () => {}
-             */
-            awaiter?: Awaiter,
-            /**
-             * Defines the root-path, where the page-objects should be stored
-             *
-             * @type {string}
-             * @default '/e2e'
-             */
-            e2eTestPath?: string,
-            /**
-             * Defines if the protractor-setting WaitForAngular should be enabled or disabled.
-             *
-             * @type {boolean}
-             * @default true
-             */
-            waitForAngularEnabled?: boolean,
-            /**
-             * Forbid hvstr to create default page object directory's.
-             *
-             * @type {boolean}
-             */
-            doNotCreateDirectorys?: boolean,
-        }
+        public options: IPageObjectBuilderOptions,
     ) {
-        this.codeBuilder = params.codeBuilder || new QueuedCodeBuilder('  ');
-        this.awaiter = params.awaiter || (async () => { });
-        this.waitForAngularEnabled = params.waitForAngularEnabled === undefined ? true : params.waitForAngularEnabled;
-        this.packagePath = new ProjectPathUtil(process.cwd(), params.e2eTestPath || '/e2e');
-        if (!params.doNotCreateDirectorys) {
+        this.options.codeBuilder = options.codeBuilder || new QueuedCodeBuilder('  ');
```

Comment:
I would change the default handling, so that you start with an initial default options object and merge the user options appropriately